src/main/scala/com/monsanto/arch/cloudformation/model/resource/EC2.scala

```diff
@@ -562,6 +562,35 @@ object `AWS::EC2::VPC` extends DefaultJsonProtocol {
   implicit val format: JsonFormat[`AWS::EC2::VPC`] = jsonFormat7(`AWS::EC2::VPC`.apply)
 }
 
+sealed trait VpcRef
+case class ResourceRefVpc(ref: Token[ResourceRef[`AWS::EC2::VPC`]]) extends VpcRef
+case class ImportedVpc(fn: `Fn::ImportValue`) extends VpcRef
+case class MappedVpc(fn: `Fn::FindInMap`[String]) extends VpcRef
+case class VpcLiteral(vpcId: String) extends VpcRef
+
+case class VpcId private(resource: VpcRef)
```

Comment:
What are you trying to prevent with the `private` here?

Reply:
The idea is that one should never have to construct a `VpcId` explicitly. I was thinking users would create instances of `VpcRef` and they would be implicitly converted to `VpcId` instances by the implicit methods below.

I modeled this off some other code in this project (although I don't remember exactly where). Idk, maybe it's cleaner to get rid of `VpcId` and use `VpcRef` directly instead? That seems to work too.

Reply:
Seems fine to me.  Just curious if there were some illegal cases you were preventing.